acme_diags/driver/tc_analysis_driver.py

```diff
@@ -0,0 +1,541 @@
+    # # of storms over the ocean)
+    result_obs: Dict[str, Any] = {}
+    result_obs["num_years"] = OBS_YEARS.size
+
+    for basin, basin_info in BASIN_DICT.items():
+        nc = netcdffile(
+            os.path.join(reference_data_path, "IBTrACS.{}.v04r00.nc".format(basin))
+        )
+
+        # Extract and parse variables from .nc file
+        vsmc = np.squeeze(nc["wmo_wind"][:, :])
+        time = np.squeeze(nc["time"][:, :])
+        monthmc, yearic = _get_monthmc_yearic(time)
+        num_rows = time.shape[0]
+
+        # Calculate mean ace
+        mean_ace = _calc_mean_ace(vsmc, yearic, num_rows)
+
+        # Calculate TS intensity distribution and seasonal cycle
+        mon, wnd = _get_mon_wind(vsmc, monthmc, yearic, num_rows)
+        tc_intensity_dist = _calc_ts_intensity_dist(wnd)
+        pdf_obs_seasonal_cycle = _calc_seasonal_cycle(mon)
+
+        # Number of hurricanes and storms over the ocean
+        num_hurricanes_per_yr = basin_info[5]
+        num_storms_over_ocean = 1
+
+        # Final result
+        result_obs[basin] = (
+            mean_ace,
+            tc_intensity_dist,
+            pdf_obs_seasonal_cycle,
+            num_hurricanes_per_yr,
+            num_storms_over_ocean,
+        )
+
+    return result_obs
+
+
+def _get_monthmc_yearic(time: "MaskedArray") -> Tuple[np.ndarray, np.ndarray]:
+    """Extracts the monthmc and yearic by parsing the time variable.
+
+    All masked (missing) data points are ignored.
+
+    :param time: Time variable, units are days since 1858-11-17 00:00:00
+    :type time: MaskedArray
+    :return: Arrays for the months and years based on the day of a hurricane
+    :rtype: Tuple[np.ndarray, np.ndarray]
+    """
+    monthmc = np.zeros(time.shape)
+    yearic = np.zeros(time.shape[0])
+
+    # The units attribute of the time variable
+    days_since = datetime(1858, 11, 17, 0, 0, 0)
+
+    for i in range(0, time.shape[0]):
+        for j in range(0, time.shape[1]):
+            if time[i, j] is not np.ma.masked:
+                day_hurr = days_since + timedelta(time[i, j])
+                monthmc[i, j] = day_hurr.month
+                yearic[i] = day_hurr.year
+
+    return monthmc, yearic
+
+
+def _get_mon_wind(
+    vsmc: "MaskedArray",
+    monthmc: np.ndarray,
+    yearic: np.ndarray,
+    num_rows: int,
+) -> Tuple[List[int], List[int]]:
+    """Extracts the months and max wind speeds.
+
+    :param vsmc: Maximum sustained wind speed from official WMO agency.
+    :type vsmc: MaskedArray
+    :param monthmc: Months
+    :type monthmc: np.ndarray
+    :param yearic: Years
+    :type yearic: np.ndarray
+    :param num_rows: Number of rows in the data matrix
+    :type num_rows: int
+    :return: Array of months and max wind speeds
+    :rtype: Tuple[List[int], List[int]]
+    """
+    mon = []
+    wnd = []
+
+    for i in range(0, num_rows):
+        if OBS_START_YR <= yearic[i] <= OBS_END_YR:
```

Comment:
Updated this if statement to be more Pythonic.

Originally `yearic[i] >= OBS_START_YR and yearic[i] <= OBS_END_YR`